### PR TITLE
[486838] fixed bug in ErrorTreeAppendable

### PR DESCRIPTION
**Pull request by sailingKieler**
Signed-off-by: Christian Schneider <christian.schneider@typefox.io>

Please review!

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=486838

**Comment by spoenemann** [comment]
As discussed earlier, the current solution uses a single leaf node, which probably works only for simple types, but not for parameterized types etc.

Don't merge yet.

**Comment by sailingKieler** [comment]
Now, fixed the issue by relying on `NodeModelUtil.getTokenText()`. Added a bunch of tests with plain types, inner types, and parameterized types.

File: plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/output/ErrorTreeAppendable.java [hunk 1]
```diff
@@ -58,7 +59,7 @@ public TreeAppendable append(JvmType type) {
 			if(unresolvedLink != null) {
 				INode linkNode = unresolvedLink.getThird();
 				if(linkNode != null) {
-					append(linkNode.getText().trim());
+					append(NodeModelUtils.getTokenText(linkNode).trim());
```
Comment (meysholdt) [inline plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/output/ErrorTreeAppendable.java]: calling .trim() is not necessary as getTokenText() returns already-trimmed strings.

**Comment by meysholdt** [comment]
besides the unnecessary call of trim() the change looks good to me

**Comment by sailingKieler** [comment]
OK, I'll drop it and merge.

**Pull request merged**